src/awkward/_slicing.py

```diff
@@ -151,9 +151,26 @@ def normalise_item(item, backend: ak._backends.Backend):
     elif isinstance(item, ak.contents.NumpyArray):
         return item.data
 
+    elif isinstance(item, ak.contents.RegularArray):
+        # Pure NumPy arrays without masks follow NumPy advanced indexing
+        # If we can produce such a content, return the underlying NumPy array
+        # Otherwise, we probably have options or are not purelist_regular, etc.
+        # As such, we then follow Awkward indexing. This logic should follow
+        # the generic `isinstance(item, ak.contents.Content)` case below
+        as_numpy = item.maybe_to_NumpyArray()
+
+        if as_numpy is None:
+            out = normalise_item_bool_to_int(normalise_item_nested(item))
+            if out.is_numpy:
+                return out.data
+            else:
+                return out
+        else:
+            return as_numpy.data
+
     elif isinstance(item, ak.contents.Content):
         out = normalise_item_bool_to_int(normalise_item_nested(item))
-        if isinstance(out, ak.contents.NumpyArray):
+        if out.is_numpy:
```

Comment:
Be careful: I think EmptyArray also has `is_numpy` (because it's coerceable). Is this what you want here?